MovieRecommender/src/main/scala/co/cask/cdap/apps/movierecommender/RecommendationBuilder.scala

```diff
@@ -26,19 +26,18 @@
 package co.cask.cdap.apps.movierecommender
 
 import co.cask.cdap.api.common.Bytes
-import co.cask.cdap.api.spark.{ScalaSparkProgram, SparkContext}
-import org.apache.hadoop.io.Text
-import org.apache.spark.SparkContext._
+import co.cask.cdap.api.spark.{SparkExecutionContext, SparkMain}
+import org.apache.spark.SparkContext
 import org.apache.spark.mllib.recommendation.{ALS, Rating}
-import org.apache.spark.rdd.NewHadoopRDD
+import org.apache.spark.rdd.RDD
 import org.slf4j.{Logger, LoggerFactory}
 
 import scala.util.control.Exception._
 
 /**
  * Spark Program which makes recommendation for movies to users
  */
-class RecommendationBuilder extends ScalaSparkProgram {
+class RecommendationBuilder extends SparkMain {
   private final val LOG: Logger = LoggerFactory.getLogger(classOf[RecommendationBuilder])
```

Comment:
This shouldn't define in the class. You need a companion object, otherwise if you try to use the logger inside closure, you'll get some strange serialization error.

```scala
class RecommendationBuilder extends SparkMain {
  import RecommendationBuilder._

  ...
}

object RecommendationBuilder {
  val LOG = LoggerFactory.getLogger(...);
}